QuasarHelios2015/src/org/team1540/quasarhelios/AutonomousModeBase.java

```diff
@@ -0,0 +1,57 @@
+package org.team1540.quasarhelios;
+
+import ccre.channel.BooleanInputPoll;
+import ccre.ctrl.BooleanMixing;
+import ccre.instinct.AutonomousModeOverException;
+import ccre.instinct.InstinctModeModule;
+
+public abstract class AutonomousModeBase extends InstinctModeModule {
+	public AutonomousModeBase(String modeName) {
+		super(modeName);
+	}
+	
+	protected void drive(float distance) throws AutonomousModeOverException, InterruptedException {
+		float startingEncoder = DriveCode.leftEncoder.get();
+		
+		if (distance > 0) {
+			DriveCode.allMotors.set(1.0f);
+			waitUntilAtLeast(DriveCode.leftEncoder, startingEncoder + distance);
+		} else {
+			DriveCode.allMotors.set(-1.0f);
+			waitUntilAtMost(DriveCode.leftEncoder, startingEncoder + distance);
+		}
+
+		DriveCode.allMotors.set(0.0f);
+	}
+	
+	protected void turn(float degree) throws AutonomousModeOverException, InterruptedException {
+		float startingYaw = HeadingSensor.yaw.get();
+
+		if (degree > 0) {
+			DriveCode.rotate.set(1.0f);
+			waitUntilAtLeast(HeadingSensor.yaw, startingYaw + degree);
+		} else {
+			DriveCode.rotate.set(-1.0f);
```

Comment:
Same here, but perhaps it should be a separate tuning, because we might want to turn slower than we drive.